casper/src/main/resources/PoS.rho

```diff
@@ -1,180 +1,165 @@
-                  paymentStore!(payment) | return!(payment)
+                for(@userPk <- userCh) {
+                  deposit!(userPk, amount, *returnCh)
                 }
               }
-
-              | contract PoS (@"pay", @amount, return) = {
-                new parametersCh, depositRet in {
-                  deposit!(amount, *depositRet)
-                  | getParameters!(*parametersCh)
-                  | for (@depositResult <- depositRet; @codeHash, @phloRate, @userPk, @timestamp <- parametersCh; lastPayment <- paymentStore) {
-                    match depositResult {
-                      (true, Nil) => { paymentStore!([codeHash, userPk, timestamp, phloRate * amount]) | return!(phloRate *amount) }
-                      (false, err) => { paymentStore!([codeHash, userPk, timestamp, 0]) | return!(0) }
-                    }
-                  }
+            }
+          }
+        } |
+
+        contract deposit (@fromUserPk, @amount, returnCh) = {
+          new vaultCh,
+              revAddressCh,
+              authKeyCh in {
+                stdout!("get deployerauthkey") |
+            @RevVault!("deployerAuthKey", *authKeyCh) |
+            for (@authKey <- authKeyCh) {
+              stdout!({"deployerauthkey" : authKey}) |
+              revAddressOps!("fromPublicKey", fromUserPk, *revAddressCh) |
+              for (@fromRevAddress <- revAddressCh) {
+                stdout!({"address": fromRevAddress}) |
+                @RevVault!("findOrCreate", fromRevAddress, *vaultCh) |
+                for (@(true, fromVault) <- vaultCh) {
+                  stdout!({"vault":fromVault}) |
+                  @fromVault!("transfer", posRevAddress, amount, authKey, *returnCh)
                 }
               }
             }
           }
         }
+      } |
+
+      contract getUser (returnCh) = {
+        new parametersCh in {
+          getParameters!(*parametersCh) |
+          for (@codeHash, @phloRate, @userPk, @timestamp <- parametersCh) {
+            returnCh!(userPk)
+          }
+        }
+      } |
+
+      contract getCurrentUserAddress(returnCh) = {
+        new userCh in {
+          getUser!(*userCh) |
+          for (@userPk <- userCh) {
+            revAddressOps!("fromPublicKey", userPk, *returnCh)
+          }
+        }
+      } |
+
+      contract getCurrentUserVault(returnCh) = {
+        new revAddressCh in {
+          getCurrentUserAddress!(*revAddressCh) |
+          for (@revAddress <- revAddressCh) {
+            @RevVault!("findOrCreate", revAddress, *returnCh)
+          }
+        }
+      }
+    } |
+
+    new uriCh in {
+      registryInsert!(
+        "cc87bf7747a8c176714b417ca14a63897b07572876c5e38a7896b6007738ef81".hexToBytes(),
+        (9223372036854775807, bundle+{*MakePoS}),
+        "7ff160accda7f76f14f2ae132aaae140ac22b2671e561a58a566bff1a01e49cf372201a5df678aba5ff09171b6af966906deffbf8933efeeddb69a4c204b0503".hexToBytes(),
+        *uriCh
+      ) |
+      for (@uri <- uriCh) {
```

Comment:
What is the point of this continuation?

Reply:
It's for debugging reasons. I had some situations when the contract was not registered in the registry and because if this the tests failed without too much clues for the poor developer. Then I started to add these continuations that print to the stdout what contracts were registered and their URIs. On a second thought, maybe I should move this log message in scala implementation of registry insertSigned.

I'll remove it though.